File: 22_09/22_09_06/boj-10974.js
const solution = (N, depth=0) =>{
    const answer = []
    const back_tracking = (arr) =>{
        if(arr.length == N){
            answer.push(arr.join(" "))
            return
        }else{
            for (let i= 1; i<= N; i++){
                if(!arr.includes(i)){
                    arr.push(i);
                    back_tracking(arr)
                    arr.pop();
                }
            }
        }
    }
    back_tracking([])
    console.log(answer)
}


solution(3)

File: 22_09/22_09_09/boj-10974.py
N = int(input())
answer = []
def back():
    if len(answer) == N:
        print(" ".join(map(str, answer))) 
        return
    for number in range(1, N+1): 
        if number not in answer:
            answer.append(number) 
            back() 
            answer.pop()
back()